graph/pull_v2.go

```diff
@@ -226,12 +240,28 @@ func (p *v2Puller) pullV2Tag(out io.Writer, tag, taggedName string) (verified bo
 			out.Write(p.sf.FormatProgress(stringid.TruncateID(img.ID), "Already exists", nil))
 			continue
 		}
+
+		digest := manifest.FSLayers[i].BlobSum
+		blobs := p.repo.Blobs(context.Background())
+		desc, err := blobs.Stat(context.Background(), digest)
+		if err != nil {
+			logrus.Debugf("Error statting layer: %v", err)
+			return false, err
+		}
```

Comment:
These `Stat`s will run even if dry-run mode is turned off, slowing down normal pulls. This should be conditioned on having dry-run mode turned on. It would be cleanest to move this to a separate function that only gets called in dry-run mode.